modules/plugin/geopkg/src/main/java/org/geotools/geopkg/GeoPackage.java

```diff
@@ -795,6 +797,38 @@ public void create(FeatureEntry entry, SimpleFeatureType schema) throws IOExcept
         entry.init(e);
     }
 
+    /**
+     * According to GeoPKG spec, the coordinates MUST be in XY order. If this FC is in YX format, we
+     * reproject to the equivalent XY project.
+     *
+     * <p>If already XY, return the input FC.
+     *
+     * @param fc underlying feature collection
+     * @return feature collection which is has axis order in XY (NORTH_EAST)
+     */
+    public static SimpleFeatureCollection forceXY(SimpleFeatureCollection fc) {
```

Comment:
This is a generic utility method (nothing specific to GPKG) and still allows to write ordinates in the wrong direction.
I was expecting to see a fix in the actual writing machinery (which I believe would require a small change in JDBCFeatureStore or JDBCDataStore along with a Dialect change to allow implementation-specific axis order forcing).

Reply:
I would also like to see a fix to JDBCDataStore createSchema as a separate PR - for now I am calling it out as a user-beware in the documentation. You are correct that such a change would require an API change (require access to the CoordinateReferenceSystem rather than just the SRID number). If you are interested in outlining fix together @aaime I would be happy to work on it with you.

Reply:
@davidblasby could you reduce this method to package visibility (so test cases can run) without committing to a new public api here.